src/devices/bus/centronics/etibuffer.cpp

```diff
@@ -0,0 +1,369 @@
+	m_bufferready_led.resolve();
+	m_ack_timer = timer_alloc(TIMER_ACK);
+	m_strobeout_timer = timer_alloc(TIMER_STROBEOUT);
+
+	save_item(NAME(m_ram));
+	save_item(NAME(m_data));
+	save_item(NAME(m_datalatch));
+	save_item(NAME(m_strobereceived));
+	save_item(NAME(m_strobe));
+	save_item(NAME(m_busy));
+}
+
+void etiprintbuffer_device::device_reset()
+{
+	address_space &mem = m_maincpu->space(AS_PROGRAM);
+	u16 ram16kblocks = ioport("CONFIG")->read() >> 2;
+	if (ram16kblocks) mem.install_ram(0x4000,0x4000+ram16kblocks*0x4000-1,m_ram);
+
+	output_busy(0);
+	output_fault(1);
+	output_ack(1);
+	output_select(1);
+	output_perror(0);
+	m_ctx->write_strobe(ASSERT_LINE);
+}
+
+//--------------------------------------------------
+//  device mconfig
+//--------------------------------------------------
+
+void etiprintbuffer_device::device_add_mconfig(machine_config &config)
+{
+	Z80(config, m_maincpu, 2_MHz_XTAL);
+	m_maincpu->set_addrmap(AS_PROGRAM, &etiprintbuffer_device::etiprintbuffer_device_memmap);
+	m_maincpu->set_addrmap(AS_IO,      &etiprintbuffer_device::etiprintbuffer_device_iomap);
+
+	CENTRONICS(config, m_ctx, centronics_devices, "printer");
+	m_ctx->busy_handler().set(FUNC(etiprintbuffer_device::busy_w));
+
+	OUTPUT_LATCH(config, m_ctx_data_out);
+	m_ctx->set_output_latch(*m_ctx_data_out);
+
+#ifdef DEBUG_ETI_BUFFER
+	// video hardware
+	screen_device &screen(SCREEN(config, m_screen, SCREEN_TYPE_RASTER));
+	screen.set_refresh_hz(60);
+	screen.set_vblank_time(ATTOSECONDS_IN_USEC(0));
+	screen.set_size(SCREEN_WIDTH, SCREEN_HEIGHT);
+	screen.set_visarea(0, SCREEN_WIDTH-1, 0, SCREEN_HEIGHT-1);
+	screen.set_screen_update(FUNC(etiprintbuffer_device::screen_update_etibuffer));
+	screen.set_physical_aspect(1,40); // long and slender
+#endif
+
+	config.set_default_layout(layout_etibuffer);
+}
+
+//--------------------------------------------------
+//  tiny rom entry
+//--------------------------------------------------
+
+
+const tiny_rom_entry *etiprintbuffer_device::device_rom_region() const
+{
+	return ROM_NAME( etiprintbuffer_device );
+}
+
+//--------------------------------------------------
+//  ioport
+//--------------------------------------------------
+
+ioport_constructor etiprintbuffer_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME( etiprintbuffer_device );
+}
+
+//--------------------------------------------------
+//  INPUT_CHANGED_MEMBERS
+//--------------------------------------------------
+
+INPUT_CHANGED_MEMBER(etiprintbuffer_device::clear_sw)
+{
+	m_maincpu->set_input_line(INPUT_LINE_NMI, newval ? CLEAR_LINE : ASSERT_LINE);
+}
+
+INPUT_CHANGED_MEMBER(etiprintbuffer_device::test_sw)
+{
+	m_maincpu->set_input_line(INPUT_LINE_IRQ0, newval ? CLEAR_LINE : ASSERT_LINE);
+}
+
+//--------------------------------------------------
+//  WRITE_LINE_MEMBERS
+//--------------------------------------------------
+
+WRITE_LINE_MEMBER( etiprintbuffer_device::busy_w )
+{
+	m_busy = (state == ASSERT_LINE) ? true : false;
```

Comment:
The Centronics stuff works with 1/0 signal levels, not assert/clear states.

Reply:
I thought that being WRITE_LINE_MEMBER meant that it would get ASSERT_LINE and CLEAR_LINE but if you'd prefer 1 and 0 that's ok.

so then: 

m_busy = (state == 1) ? true : false;